tests/results/DiffTest.py

```diff
@@ -628,15 +630,30 @@ def test_delete(self):
         self.uut.delete = False
 
     def test_add_linebreaks(self):
+        expected = ['1\n', '2\n', '3\n']
+
         self.assertEqual(
             Diff._add_linebreaks(['1', '2', '3']),
-            ['1\n', '2\n', '3\n'])
+            expected)
 
         self.assertEqual(Diff._add_linebreaks([]), [])
 
+        self.assertEqual(Diff._add_linebreaks(expected), expected)
+
     def test_generate_linebreaks(self):
+        no_lb_eof = ['1\n', '2\n', '3']
+        lb_eof = ['1\n', '2\n', '3\n']
+
         self.assertEqual(
             Diff._generate_linebreaks(['1', '2', '3']),
-            ['1\n', '2\n', '3'])
+            no_lb_eof)
 
         self.assertEqual(Diff._generate_linebreaks([]), [])
+
+        self.assertEqual(
+            Diff._generate_linebreaks(no_lb_eof),
+            no_lb_eof)
+
+        self.assertEqual(
+            Diff._generate_linebreaks(lb_eof),
+            lb_eof)
```

Comment:
I think those changes should go into the other commit introducing `generate_linebreaks` and `add_linebreaks`

(GitHub does display the same comments in different commits if they touch the same parts of code, sadly... not really fitting our review style^^)